addon/components/polaris-resource-list/bulk-actions.js

```diff
@@ -0,0 +1,535 @@
+      return [{ items: actionsCollection }];
+    }
+  }),
+
+  combinedActions: computed(
+    'actionSections.[]',
+    'rolledInPromotedActions',
+    function() {
+      let combinedActions = [];
+      let { actionSections, rolledInPromotedActions } = this.getProperties(
+        'actionSections',
+        'rolledInPromotedActions'
+      );
+
+      let rolledInPromotedActionsHasLength = rolledInPromotedActions.length > 0;
+
+      if (actionSections && rolledInPromotedActionsHasLength) {
+        combinedActions = [
+          { items: rolledInPromotedActions },
+          ...actionSections,
+        ];
+      } else if (actionSections) {
+        combinedActions = actionSections;
+      } else if (rolledInPromotedActionsHasLength) {
+        combinedActions = [{ items: rolledInPromotedActions }];
+      }
+
+      return combinedActions;
+    }
+  ),
+
+  /**
+   * Callback when selectable state of list is changed
+   *
+   * @type {Function}
+   * @default noop
+   * @property onSelectModeToggle
+   */
+  onSelectModeToggle(/* selectMode **/) {},
+
+  /**
+   * Callback when the select all checkbox is clicked
+   *
+   * @type {Function}
+   * @default noop
+   * @property onToggleAll
+   */
+  onToggleAll() {},
+
+  instanceOfBulkActionListSectionArray(actionsCollection) {
+    let validList = actionsCollection.filter((action) => {
+      return action.items;
+    });
+
+    return actionsCollection.length === validList.length;
+  },
+
+  instanceOfBulkActionArray(actionsCollection) {
+    let validList = actionsCollection.filter((action) => {
+      return !action.items;
+    });
+
+    return actionsCollection.length === validList.length;
+  },
+
+  setContainerNode() {
+    this.set('containerNode', this.element);
+  },
+
+  setMoreActionsNode(node) {
+    this.set('moreActionsNode', node);
+  },
+
+  setLargeScreenButtonsNode(node) {
+    this.set('largeScreenButtonsNode', node);
+  },
+
+  handleResize() {
+    throttle(
+      this,
+      function() {
+        let {
+          smallScreenPopoverVisible,
+          largeScreenPopoverVisible,
+          containerNode,
+        } = this.getProperties(
+          'smallScreenPopoverVisible',
+          'largeScreenPopoverVisible',
+          'containerNode'
+        );
+
+        if (containerNode) {
+          let containerWidth = containerNode.getBoundingClientRect().width;
+          if (containerWidth > 0) {
+            this.set('containerWidth', containerWidth);
+          }
+        }
+
+        if (smallScreenPopoverVisible || largeScreenPopoverVisible) {
+          this.setProperties({
+            smallScreenPopoverVisible: false,
+            largeScreenPopoverVisible: false,
+          });
+        }
+      },
+      50
+    );
+  },
+
+  didInsertElement() {
+    this._super(...arguments);
+
+    let promotedActions = this.get('promotedActions');
+
+    if (promotedActions && promotedActions.length > MAX_PROMOTED_ACTIONS) {
+      warn(
```

Comment:
`warn` [needs](https://github.com/emberjs/ember.js/blob/v3.5.1/packages/%40ember/debug/lib/warn.ts#L102) to be given an `id` for the warning otherwise this will log a deprecation warning. We're already doing this in e.g. [`polaris-popover`](https://github.com/smile-io/ember-polaris/blob/master/addon/components/polaris-popover.js#L167) so just follow suit from that 👍